clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/ReservationReportCachingAgent.groovy

```diff
@@ -110,90 +117,105 @@ class ReservationReportCachingAgent implements CachingAgent {
 
   @Override
   CacheResult loadData(ProviderCache providerCache) {
+    long startTime = System.currentTimeMillis()
     log.info("Describing items in ${agentType}")
-    Map<String, AmazonReservationReport.OverallReservationDetail> reservations = [:].withDefault { String key ->
+
+    Map<String, AmazonReservationReport.OverallReservationDetail> reservations = new ConcurrentHashMap<>()
+    def getReservation = { String key ->
```

Comment:
both places this is called it is assembling the key out of the component parts to call it .. could parameterize the closure with the parts, and construct the key inside here